burikova_anna/graph_print.cpp

```diff
@@ -0,0 +1,155 @@
+#include <fstream>
+#include <iostream>
+#include <iterator>
+#include <string>
+#include <vector>
+
+int kVerticesCount = 14;
+
+class Graph {
+ public:
+  using VertexId = int;
+  using EdgeId = int;
+  struct Vertex {
+   public:
+    explicit Vertex(VertexId id) : id_(id) {}
+    VertexId id() const { return id_; }
+
+   private:
+    VertexId id_ = 0;
+  };
+  struct Edge {
+   public:
+    Edge(EdgeId id, VertexId from_vertex_id, VertexId to_vertex_id)
+        : id_(id),
+          from_vertex_id_(from_vertex_id),
+          to_vertex_id_(to_vertex_id) {}
+    EdgeId id() const { return id_; }
+    VertexId from_vertex_id() const { return from_vertex_id_; }
+    VertexId to_vertex_id() const { return to_vertex_id_; }
+
+   private:
+    EdgeId id_ = 0;
+    VertexId from_vertex_id_ = 0;
+    VertexId to_vertex_id_ = 0;
+  };
+  std::vector<Vertex> get_vertices() const { return vertices_; }
+  std::vector<Edge> const get_edges() const { return edges_; }
```

Comment:
1) Ты в этих функциях возвращаешь полную копию данных, это очень неэффективно. Лучше будет вернуть `const` ссылку на данные. Таким образом мы избежим копирования и обеспечим безопасность.
2) Возвращать `const` копию даных смысла нет, так как это копия и нас не беспокоит, если её кто-то изменит.
```suggestion
  const std::vector<Vertex>& get_vertices() const { return vertices_; }
  const std::vector<Edge>& get_edges() const { return edges_; }
```